### PR TITLE
PLT-1459: Updated README of marconi-chain-index and marconi-sidechain

### PR DESCRIPTION
**Pull request by koslambrou**
* Rename the old `marconi-chain-index` README.md to README_MARCONI_DRAFT as it represents the README for the Marconi project as a whole. Will probably be used elsewhere.

* Update the `marconi-sidechain` README with latest changes and added some instructions on how to build it.

* Create a README for `marconi-chain-index` with a similar format to `marconi-sidechain` as they share many similar things. This will be worked on more in the future.

Note: The diff in the `marconi-chain-index/README.md` can be glanced over very quickly, as it's basically the same as the `marconi-sidechain/README.md` (minus the JSON-RPC server)

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [ ] Commit sequence broadly makes sense
    - [ ] Key commits have useful messages
    - [ ] Formatting, PNG optimization, etc. are updated
    - [ ] Important changes are reflected in changelog.d of the affected packages
- PR
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] Reference the ADR in the PR and reference the PR in the ADR (if revelant)
    - [ ] Reviewer requested

### CODE REVIEW

File: marconi-sidechain/README.md [hunk 1]
```diff
@@ -29,173 +28,199 @@ The interface for marconi-sidechain uses [JSON-RPC](http://www.simple-is-better.
...
-                           address-2 ..."
-  -h,--help                Show this help text
-```
+We are assuming that:
 
-## Example of using marconi-sidechain
+* you have a local running cardano-node instance.
+* you've set the following environment variables:
+  * `CARDANO_NODE_SOCKET_PATH`: socket path of your local cardano-node instance
+  * `MARCONI_DB_DIRECTORY`: directoy in which to create the various SQLite database files
```
Comment (kayvank) [inline marconi-sidechain/README.md:100]: direcory

File: marconi-chain-index/README.md [hunk 2]
```diff
@@ -1,178 +1,99 @@
...
-## Architecture Decision Records
+The most minimal way to run the executable is as follows:
 
-* Records of decisions that were made by the team.
-* Discuss alternatives.
-* Why did we choose one design over another?
-   * Konstantinos has some for Plutus Apps.
-   * Radu has some for Marconi.
+```sh
+marconi-chain-index \
```
Comment (eyeinsky) [inline marconi-chain-index/README.md:77]: I'm not a fan of the golden tests that we have for the cli, but now that we do have them let's link to them?

File: marconi-sidechain/README.md [hunk 3]
```diff
@@ -29,173 +28,199 @@ The interface for marconi-sidechain uses [JSON-RPC](http://www.simple-is-better.
...
-The following is an example shell script for executing marconi-sidechain in [preview-testnet](https://book.world.dev.cardano.org/environments.html#preview-testnet).
+* from the last chainpoint (if none, from genesis), fetch blocks from the local node, extract required data and index them in the database.
+* run a JSON-RPC server which will listen for any queries on the indexed data.
 
-### Requirement
+Using the `--addresses-to-index`, you can instruct `marconi-sidechain` to index target addresses.
+By default, all addresses are indexed in the database.
 
-Compatible version of [cardano-node](https://github.com/input-output-hk/plutus-apps/blob/main/cabal.project#L246) must be running with the socket-path as outlined below:
+Some example addresses you can for pre-production-testnet are:
```
Comment (eyeinsky) [inline marconi-sidechain/README.md:119]: you can => you can use

**Comment by eyeinsky** [comment]
I found the readme that is now renamed to draft useful when I first joined as it gave an overview of what marconi did. While the new parts on how to build and run is also useful, then maybe they could be merged instead -- as the new readme is brief on what marconi does, what it's architecture is, etc.

File: marconi-chain-index/README.md [hunk 4]
```diff
@@ -1,178 +1,99 @@
...
+    --socket-path "$CARDANO_NODE_SOCKET_PATH" \
+    --db-dir "$MARCONI_DB_DIRECTORY" \
+```
 
-## System Requirements
+This command will do two things:
 
-## How to Install and Configure
+* from the last chainpoint (if none, from genesis), fetch blocks from the local node, extract required data and index them in the database.
+* run a JSON-RPC server which will listen for any queries on the indexed data.
```
[the remark this answers is not in the document]
Comment (koslambrou) [reply marconi-chain-index/README.md:86]: Copy paste error. Thanks :)

**Comment by koslambrou** [comment]
> I found the readme that is now renamed to draft useful when I first joined as it gave an overview of what marconi did. While the new parts on how to build and run is also useful, then maybe they could be merged instead -- as the new readme is brief on what marconi does, what it's architecture is, etc.

@eyeinsky I agree that it's very valuable. However, it is **not** the README for `marconi-chain-index`, but README for the Marconi project as a whole. That draft README would essentially be the README for the marconi project once we split it in a separate repo.

File: marconi-chain-index/README.md (hunk 2 above)
Comment (koslambrou) [reply marconi-chain-index/README.md:77]: We don't have golden tests for the `marconi-chain-index` CLI, and it's a bit different than the `marconi-sidechain` one.

Comment (koslambrou) [reply marconi-chain-index/README.md:77]: I made an argument here https://github.com/input-output-hk/plutus-apps/pull/960#discussion_r1109924411 about why we would want golden tests.

What don't you like exactly? What can we do different which can achieve the same benefits as the golden tests that we have.

Comment (kayvank) [reply marconi-chain-index/README.md]: > We don't have golden tests for the `marconi-chain-index` CLI, and it's a bit different than the `marconi-sidechain` one.

We added CLI golden test to [marconi-shain-index ](https://github.com/input-output-hk/plutus-apps/blob/main/marconi-chain-index/test/Spec/Golden/Cli/marconi-chain-index___help.help)

Comment (koslambrou) [reply marconi-chain-index/README.md]: @kayvank Thanks, added a link to the golden file. However, in this case, I want to show an example off how to start the command in the README, and the golden file doesn't provide that. We can probably extend it, but for now, let's keep it as is.

Comment (eyeinsky) [reply marconi-chain-index/README.md]: Regarding not liking golden tests for CLI: the CLI text that an executable prints are directly generated from the CLI definitions that we defined by the help of optparse-applicative library and tests for that process already exist in optparse-applicative. I.e if we don't want to change the CLI then we should just ...not change the CLI definitions. And when we *do* want to change the CLI definitions then we now have *two* files to change.

**Review by eyeinsky — APPROVED** [review]

**Pull request merged**